addons/l10n_hk/models/res_bank.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import re
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools import single_email_re
+
+
+auto_mobn_re = re.compile(r"""^[+]\d{1,3}-\d{1,29}$""", re.VERBOSE)
+
+class ResPartnerBank(models.Model):
+    _inherit = 'res.partner.bank'
+
+    country_code = fields.Char(related='partner_id.country_code', string='Country Code')
+    l10n_hk_fps_type = fields.Selection([('id', 'FPS ID'), ('phone', 'Phone Number'), ('email', 'Email Address')], string='FPS Type', default='id')
+    l10n_hk_fps_identifier = fields.Char(string='FPS ID/Phone Number/Email Address')
+
+    @api.constrains('l10n_hk_fps_type', 'l10n_hk_fps_identifier')
+    def _check_l10n_hk_fps_identifier(self):
+        for bank in self:
+            if bank.l10n_hk_fps_type == 'id' and (not bank.l10n_hk_fps_identifier or len(bank.l10n_hk_fps_identifier) not in [7, 9]):
+                raise ValidationError(_('Invalid FPS ID! Please enter a valid FPS ID with length 7 or 9.'))
+            if bank.l10n_hk_fps_type == 'phone' and (not bank.l10n_hk_fps_identifier or not auto_mobn_re.match(bank.l10n_hk_fps_identifier)):
+                raise ValidationError(_('Invalid Mobile! Please enter a valid mobile number with format +852-67891234.'))
+            if bank.l10n_hk_fps_type == 'email' and (not bank.l10n_hk_fps_identifier or not single_email_re.match(bank.l10n_hk_fps_identifier)):
+                raise ValidationError(_('Invalid Email! Please enter a valid email address.'))
+
+    # CRC16 calculation with polynomial 0x1021 and initial value 0xFFFF
+    def _get_crc16(self, data, poly=0x1021, init=0xFFFF):
+        crc = init
+        for byte in data:
+            crc = crc ^ (byte << 8)
+            for _i in range(8):
```

Comment:
```suggestion
            for _ in range(8):
```
should be sufficient I think

Reply:
I need to `import _ from odoo` for translation, I can either change the name of the translation method or I change the name in the for loop. Which way do you prefer? Otherwise ci/style will show error message.

Reply:
Oh right ! It's okay like that then :)